[WebGPU] Convert textured-cube-shader-constant demo to WGSL

tadeuzagallo opened the pull request:
#### 2eaefed75c74c1c9430970119306a137da708b14
<pre>
[WebGPU] Convert textured-cube-shader-constant demo to WGSL
<a href="https://bugs.webkit.org/show_bug.cgi?id=255594">https://bugs.webkit.org/show_bug.cgi?id=255594</a>
rdar://108195733

Reviewed by Mike Wyrzykowski.

Convert the shaders in the demo from using MSL directly to use WGSL instead

* Websites/webkit.org/demos/webgpu/scripts/textured-cube-shader-constants.js:

Canonical link: <a href="https://commits.webkit.org/263314@main">https://commits.webkit.org/263314@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9f24b6a2ea44f27a4676523903e6e9c1b19f7761

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3691 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3775 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3881 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/5117 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/3974 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/3852 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3780 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/3196 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3735 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/3963 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/3326 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4951 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/1469 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/3296 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/3360 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/3277 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/3355 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/4722 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3744 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3031 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/3278 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/3296 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1037 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/3310 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/3555 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski commented on Websites/webkit.org/demos/webgpu/scripts/textured-cube-shader-constants.js:
```diff
@@ -100,7 +100,7 @@ async function helloCube() {
         dimension: "2d",
         format: "bgra8unorm",
-        usage: GPUTextureUsage.TEXTURE_BINDING | GPUTextureUsage.COPY_DST
+        usage: GPUTextureUsage.TEXTURE_BINDING | GPUTextureUsage.COPY_DST | GPUTextureUsage.RENDER_ATTACHMENT
```
GPUTextureUsage.RENDER_ATTACHMENT shouldn't be needed, since texture or its views are not bound as render targets in this file as far as I could see

tadeuzagallo committed:
[WebGPU] Convert textured-cube-shader-constant demo to WGSL
https://bugs.webkit.org/show_bug.cgi?id=255594
rdar://108195733

Reviewed by Mike Wyrzykowski.

Convert the shaders in the demo from using MSL directly to use WGSL instead

* Websites/webkit.org/demos/webgpu/scripts/textured-cube-shader-constants.js:

Canonical link: https://commits.webkit.org/263314@main
